matchmaking/protocol.go

```diff
@@ -55,5 +57,17 @@ func NewCommonMatchMakingProtocol(server *nex.Server) *CommonMatchMakingProtocol
 	commonMatchMakingProtocol.UpdateSessionHostV1(updateSessionHostV1)
 	commonMatchMakingProtocol.UpdateSessionHost(updateSessionHost)
 
+	if server.PRUDPVersion() == 0 {
+		server.On("Kick", func(packet *nex.PacketV0) {
+			fmt.Println("Leaving")
+			common_globals.RemoveConnectionIDFromAllSessions(packet.Sender().ConnectionID())
+		})
+	} else {
+		server.On("Kick", func(packet *nex.PacketV1) {
+			fmt.Println("Leaving")
+			common_globals.RemoveConnectionIDFromAllSessions(packet.Sender().ConnectionID())
+		})
+	}
```

Comment:
```suggestion
	server.On("Kick", func(packet nex.PacketInterface) {
		fmt.Println("Leaving")
		common_globals.RemoveConnectionIDFromAllSessions(packet.Sender().ConnectionID())
	})
```

We can use PacketInterface here